pkgs/shells/fish/default.nix

```diff
@@ -173,63 +191,69 @@ let
     # Fix FHS paths in tests
     postPatch =
       ''
-        # src/fish_tests.cpp
-        sed -i 's|/bin/ls|${coreutils}/bin/ls|' src/fish_tests.cpp
-        sed -i 's|is_potential_path(L"/usr"|is_potential_path(L"/nix"|' src/fish_tests.cpp
-        sed -i 's|L"/bin/echo"|L"${coreutils}/bin/echo"|' src/fish_tests.cpp
-        sed -i 's|L"/bin/c"|L"${coreutils}/bin/c"|' src/fish_tests.cpp
-        sed -i 's|L"/bin/ca"|L"${coreutils}/bin/ca"|' src/fish_tests.cpp
-        # disable flakey test
-        sed -i '/{TEST_GROUP("history_races"), history_tests_t::test_history_races},/d' src/fish_tests.cpp
+        sed -i 's|"/bin/ls"|"${lib.getExe' coreutils "ls"}"|' src/builtins/tests/test_tests.rs
```

Comment:
Could we use `substituteInPlace --fail` for all these simple `sed -i`? So we can assert their presences and avoid escaping trap.

Reply:
That would be better indeed, I didn't try, but it seems like we could.
I'll give that a try!

Reply:
I pushed a commit with this, it seems to build fine and the tests pass, I'm still making sure that we didn't miss anything.

I also removed some replacements that were not accurate anymore.